#!/usr/bin/env python
PACKAGE = "goal_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("camera_angle", double_t, 0, "Angle up of the camera from horizontal (degrees)", 0, -180,   180)
gen.add("blue_scale", double_t, 0, "% of blue intesity to filter out", 0, 0, 1)
gen.add("red_scale", double_t, 0, "% of red intesity to filter out", 0, 0, 1)
gen.add("min_confidence", double_t, 0, "minimum confidence value for valid targets", 0, 0, 1)
gen.add("otsu_threshold", int_t, 0, "Threshold for assuming frame has no green targets - higher means filter out brighter frames", 0, 0, 255)

exit(gen.generate(PACKAGE, "goal_detection", "GoalDetection"))
